lib/xain/helpers.ex

```diff
@@ -37,6 +38,18 @@ defmodule Xain.Helpers do
   def pop!([item | rest]) do
     {rest, item |> Enum.reverse}
   end
+
+  def ensure_valid_contents(contents, _) when is_binary(contents) or is_list(contents) do
+    contents
+  end
+  def ensure_valid_contents(contents, tag_name) when is_number(contents) or is_atom(contents) do
+    Logger.debug "#{tag_name} has been called as #{tag_name}(#{inspect(contents)}, ...), but the first argument supposed to be a binary"
+    to_string(contents)
+  end
+  def ensure_valid_contents(contents, tag_name) do
+    Logger.warn "#{tag_name} has been called as #{tag_name}(#{inspect(contents)}, ...), but the first argument supposed to be a binary"
+    inspect(contents)
```

Comment:
Thanks!!

Hijacking the conversation - Moving ex_admin over to the new xain is pretty challenging. I’m part way through it. One of the issues I’m facing is with the safe callback being run for each do markup. I’ve had to add a bunch of them. I think the there are going to be a lot more markup calls that we don’t want to run safe callback, so I’m prototyping a markup safe: true do to run the callback. So, markup do won’t run it.  What do you think?

> On May 27, 2016, at 3:24 PM, Roman Smirnov <notifications@github.com> wrote:
> 
> In lib/xain/helpers.ex <https://github.com/smpallen99/xain/pull/6#discussion_r64955454>:
> 
> > @@ -37,6 +38,18 @@ defmodule Xain.Helpers do
> >    def pop!([item | rest]) do
> >      {rest, item |> Enum.reverse}
> >    end
> > +
> > +  def ensure_valid_contents(contents, _) when is_binary(contents) or is_list(contents) do
> > +    contents
> > +  end
> > +  def ensure_valid_contents(contents, tag_name) when is_number(contents) or is_atom(contents) do
> > +    Logger.debug "#{tag_name} has been called as #{tag_name}(#{inspect(contents)}, ...), but the first argument supposed to be a binary"
> > +    to_string(contents)
> > +  end
> > +  def ensure_valid_contents(contents, tag_name) do
> > +    Logger.warn "#{tag_name} has been called as #{tag_name}(#{inspect(contents)}, ...), but the first argument supposed to be a binary"
> > +    inspect(contents)
> Well, I'm not sure that we should encourage such usage (with implicit to_string)
> But I guess, this is not so crucial question. So, I don't mind of your implementation.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/smpallen99/xain/pull/6/files/d413bacf14e7e6dae504ed3ca15c3758ca8437ff#r64955454>, or mute the thread <https://github.com/notifications/unsubscribe/AA2YpR3FktWobWkqniOoauSN1--x21Biks5qF0TzgaJpZM4Iok8i>.
>